DesignSystem/Sources/Modal/MOITAlarmView.swift

```diff
@@ -0,0 +1,202 @@
+        super.layoutSubviews()
+        self.flexRootView.pin.all()
+        self.flexRootView.flex.layout()
+        self.configureGradient()
+    }
+}
+
+// MARK: - Private functions
+
+extension MOITAlarmView {
+    
+    private func configureLayouts() {
+        self.addSubview(self.flexRootView)
+        
+        self.flexRootView.flex
+            .direction(.column)
+            .define { flex in
+                
+                flex.addItem(self.titleLabel)
+                    .marginTop(20)
+                    .marginHorizontal(16)
+                
+                flex.addItem()
+                    .direction(.row)
+                    .alignItems(.center)
+                    .define { flex in
+                        flex.addItem()
+                            .define { flex in
+                                flex.addItem(self.descriptionLabel)
+                                flex.addItem(self.mainLabel)
+                            }
+                            .grow(1)
+                        
+                        flex.addItem(self.imageView)
+                            .alignSelf(.end)
+                            .shrink(1)
+                            .aspectRatio(200/130)
+                    }
+                    .marginHorizontal(16)
+                    .height(130)
+                
+                flex.addItem(self.button)
+                    .marginHorizontal(16)
+                    .marginBottom(19)
+            }
+            .width(335)
+            .height(260)
+    }
+    
+    private func configureGradient() {
+        let gradient = CAGradientLayer()
+        gradient.colors = [
+            ResourceKitAsset.Color.black.color.cgColor,
+            UIColor.init(red: 34/255, green: 37/255, blue: 93/255, alpha: 1.0).cgColor
+        ]
+        
+        gradient.locations = [0.0, 1.0]
+        gradient.startPoint = CGPoint(x: 0.5, y: 0.0)
+        gradient.endPoint = CGPoint(x: 0.5, y: 1.0)
+        gradient.frame = self.flexRootView.bounds
+        self.flexRootView.layer.insertSublayer(gradient, at: 0)
+    }
+    
+    private func subscribeRemainTimer() {
+        Observable<Int>.interval(
+            .seconds(1),
+            scheduler: ConcurrentDispatchQueueScheduler.init(qos: .background)
+        )
+        .filter { [weak self] second in
+            guard let self = self else { return false }
+            return self.type == .출석체크 &&
+            60*3 >= second
+        }
+        .map { second -> String in
+            let remainSecond = 60*3 - second
+            let minute = remainSecond / 60
+            let second = remainSecond % 60
+            if second < 10 { return "\(minute):0\(second)" }
+            else { return "\(minute):\(second)" }
+        }
+        .observe(on: MainScheduler.instance)
+        .bind(onNext: { [weak self] remainTime in
+            self?.mainLabel.text = remainTime
+            self?.mainLabel.flex.markDirty()
+            self?.layoutIfNeeded()
```

Comment:
![스크린샷 2023-06-07 오전 8 57 51](https://github.com/mash-up-kr/MOIT-iOS/assets/89574881/67efc989-b484-4d24-9375-00b38ae93afd)
disposed 안되는 것 같습니다! 
```suggestion
        .take(while: {$0 != "0:00"} )
        .observe(on: MainScheduler.instance)
        .bind(onNext: { [weak self] remainTime in
            self?.mainLabel.text = remainTime
            self?.mainLabel.flex.markDirty()
            self?.layoutIfNeeded()
```
`take(while:)`로 설정해주면 
![스크린샷 2023-06-07 오전 9 08 30](https://github.com/mash-up-kr/MOIT-iOS/assets/89574881/42960ea7-3445-40b8-b3c2-9a1d9f7fb793)

정상적으로 disposed 됩니다 :)

timer가 리소스가 많이 드는 작업이라 확인해봤습니당

Reply:
헉 viewcontroller가 deinit될 때 타이머 dispose되는거까지만 확인했네용.. 이 부분 같이 수정하겠습니다 ~ 👍